layouts/index.json

```diff
@@ -6,17 +6,19 @@
         {{ $ancestors = $ancestors | append $a.LinkTitle }}
     {{- end -}}
 
-    {{-
-        $index = $index | append (
-            dict
-                "objectID" $page.File.UniqueID
-                "title" $page.Title
-                "section" $page.Section
-                "href" $page.RelPermalink
-                "params" $page.Params
-                "kind" $page.Kind
-                "ancestors" (after 1 $ancestors)
-        )
-    -}}
+    {{- if ne $page.File nil -}}
```

Comment:
Turns out the only pages that have nil `File` references are the tag and author pages, whose `.UniqueID` values already resolve to `""` (so are currently being filtered out anyway). So keeping only on those pages that contain `File` references looks like it's fine after all.